src/engine/compiler/MacroAssembler.v3

```diff
@@ -302,6 +304,24 @@ class MacroAssembler(valuerep: Tagging, regConfig: RegConfig) {
 
 	def emit_debugger_breakpoint();
 
+	// Gets the number of parameters to a continuation.
+	def emit_get_params_length(cont: Reg, dest: Reg);
+	// stk.state_ = state;
+	def emit_set_stack_state(stk: Reg, state: StackState);
+	// Validates {cont} and set {cont.used} to {true};
+	def emit_validate_and_consume_cont(cont: Reg);
+	// This should be called after all stack value transfers, but before the
+	// return address is pushed onto parent's {sp}.
+	// Chains {cont} on {parent} by:
+	// - setting the state of both {parent} and {cont}
+	// - setting {cont.parent} and {cont.parent_rsp_ptr}
+	// Destructive on {parent}.
+	def emit_chain_cont_to_parent(parent: Reg, cont: Reg);
+	// Called as the last step during resume. Switches {curStack} to the top of {cont}.
+	def emit_switch_curStack_to_cont(cont: Reg);
+	def emit_set_rsp_vsp(stack: Reg, rsp: Reg, vsp: Reg);
```

Comment:
I didn't see why these combine two reads and writes. Can we make them separate and following the convention I mentioned above?

Reply:
Can the explicit methods for field access be used for setting fields? I think currently there are only getters in the `MacroAssembler` refactor.